common/persistence/nosql/nosqlplugin/cassandra/gocql/client.go

```diff
@@ -51,7 +51,13 @@ func RegisterClient(c Client) {
 func newCassandraCluster(cfg ClusterConfig) *gocql.ClusterConfig {
 	hosts := parseHosts(cfg.Hosts)
 	cluster := gocql.NewCluster(hosts...)
-	cluster.ProtoVersion = 4
+
+	// Keeping compatibility
+	if cfg.ProtoVersion == 0 {
+		cfg.ProtoVersion = 4
+	}
+
+	cluster.ProtoVersion = cfg.ProtoVersion
```

Comment:
I think the if statement should be moved after the assignment.

Reply:
If I add it later `cluster` wouldn't have the right value afaik

Reply:
Oh, Sorry. You are right, I misread the code.